cmd/oci/download.go

```diff
@@ -0,0 +1,242 @@
+package download
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/konflux-ci/qe-tools/pkg/oci"
+	"github.com/spf13/cobra"
+)
+
+// downloadOptions holds the configuration options for the download command.
+// It contains flags that determine how artifacts are downloaded from OCI storage.
+type downloadOptions struct {
+	// repo specifies a single OCI repository and tag from which to download artifacts.
+	// This is used when the user wants to download from a specific repository.
+	repo string
+
+	// repos is a slice of strings representing multiple OCI repositories.
+	// This allows the user to specify a list of repositories to download from.
+	// This option should be used in conjunction with the `--since` flag to filter artifacts.
+	repos []string
+
+	// since specifies a time range for downloading the latest artifacts.
+	// It accepts durations in various formats (e.g., "4h", "2d").
+	// This flag is required when downloading from multiple repositories specified in the `repos` field.
+	since string
+
+	// ociCache specifies the directory where OCI artifacts will be cached.
+	// If not provided, a default directory will be created at $HOME/.config/qe-tools/cache.
+	ociCache string
+
+	// artifactsOutput specifies the output path for downloaded artifacts.
+	// This field is mandatory and specifies where the downloaded artifacts should be stored.
+	artifactsOutput string
+
+	// noCache determines whether to remove the OCI cache after downloading artifacts.
+	// If true, the cache will be deleted after the command execution completes, regardless of success or failure.
+	noCache bool
+}
+
+var opts = &downloadOptions{}
+
+// downloadCmd represents the download command
+var downloadCmd = &cobra.Command{
+	Use:   "download",
+	Short: "Download artifacts from OCI storage",
+	Long: `Download artifacts from OCI storage based on the specified repository or repositories.
+
+Examples:
+  - Download from a single repository:
+      qe-tools download --repo quay.io/test/test:1.0 --artifacts-output /path/to/output
+
+  - Download from multiple repositories with a time range:
+      qe-tools download --repos quay.io/repo1 quay.io/repo2 --since 4h --artifacts-output /path/to/output
+`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		log.SetFlags(log.Ldate | log.Ltime | log.Lshortfile)
+
+		// Validation: Fail if both 'repo' and 'repos' are provided
+		if opts.repo != "" && len(opts.repos) > 0 {
+			return fmt.Errorf("you cannot use both --repo and --repos at the same time")
+		}
+
+		// Validation: Fail if 'since' is provided without 'repos', or 'repos' is provided without 'since'
+		if opts.since != "" && len(opts.repos) == 0 {
```

Comment:
can we allow combination of `--repo` and `--since` flags too?

Reply:
I was thinking to have --repo for repo:tag and if you want a single repo you can use --repos quay.io/test/test1